executor.go

```diff
@@ -275,6 +294,130 @@ func (e *Executor) provideEventLogger(t *Task) *Logger {
 	}
 }
 
+func (e *Executor) parseTaskFlagsIntoMap(taskName string, flags []string) map[string]FlagArg {
+	taskFlagsMap := make(map[string]FlagArg)
+
+	for _, flag := range flags {
+		var key string
+		var val string
+		splitFlag := strings.Split(flag, "=")
+
+		key, err := e.getVerifiedFlagKey(taskName, splitFlag[0])
+		if err != nil {
+			panic(fmt.Sprintf("Unsupported flag passed to %s: `%s`. See error: %s", taskName, key, err))
+		}
+
+		if len(splitFlag) > 2 || len(splitFlag) == 0 {
+			// If the passed flag has >1 "=" in it or the arg was an empty string,
+			// then the flag has invalid syntax.
+			panic(fmt.Sprintf("Invalid flag syntax for %s: `%s`", taskName, flag))
+		} else if len(splitFlag) == 2 {
+			// If the passed flag has an "=" in it, assume that variable flag was set
+			// (e.g. --var="val").
+			val = splitFlag[1]
+		}
+
+		// At this point, we should have verified that this flag is supported in `getVerifiedFlagKey`.
+		taskFlag := e.taskFlagsRegistry[taskName][key]
+
+		// If a variable flag arg was not passed and a default exists, use the default.
+		if val == "" && taskFlag.Default != "" {
```

Comment:
how does this work with boolean flags? if my boolean flag `a` has default value `false` and i pass in `-a` instead of `-a=true`, wouldn't this set the value to `false` ?

Reply:
ok i see the separate handling of boolean flags below but it feels odd to me that we provide an incorrect Value in the flagArg below. Probably it doesn't matter for 99% of cases, but if I was writing a task I would assume that the raw Value would match what was actually passed in.

Reply:
Actually, following up let's support Defaults for booleans and I'll fix this logic cause I'd like to mimic the flag package as closely as possible:
https://pkg.go.dev/flag#Bool